### PR TITLE
Stub new task inbox endpoint

### PR DESCRIPTION
**Pull request by alexcu**
Adds endpoint for task inbox. Stubs data as applicable. Added `TODO: JAKE - <msg>` for areas that need real data for @jakerenzella to do as discussed with @macite.

### CODE REVIEW

File: app/models/task.rb [hunk 1]
```diff
@@ -184,6 +184,30 @@ def days_overdue
     (project.reference_date - task_definition.target_date).to_i / 1.day
   end
 
+  # Action date defines the last time a task has been "actioned", either the
+  # submission date or latest student comment -- whichever is newer
+  def action_date
+    return nil if last_student_comment.nil? || submission_date.nil?
+    return last_student_comment if !last_student_comment.nil? && submission_date.nil?
```
Comment (macite) [inline app/models/task.rb]: Shouldn't this return the date of that comment?

File: app/models/task.rb [hunk 2]
```diff
@@ -184,6 +184,30 @@ def days_overdue
...
+    return submission_date      if !submission_date.nil? && last_student_comment.nil?
+    last_student_comment > submission_date ? last_student_comment : submission_date
```
Comment (macite) [inline app/models/task.rb]: Similar here?

File: app/models/task.rb [hunk 3]
```diff
@@ -184,6 +184,30 @@ def days_overdue
...
+  end
+
+  # Returns the last tutor comment for this task
+  def last_tutor_comment
+    comments.where(user: project.tutorial.tutor.user).order(:created_at).last
+  end
+
+  # Returns the number of new comments for a user
+  def new_comments_for_user(user)
+    comments.map { |c| c.new_for_user?(user) }.compact.length
```
Comment (macite) [inline app/models/task.rb]: I suggest doing this with a query. Use where to limit tasks to those where recipient_id is user.id, and then count to get the count of these.

File: app/serializers/task_serializer.rb [hunk 4]
```diff
@@ -3,20 +3,26 @@ class ShallowTaskSerializer < ActiveModel::Serializer
...
+             :quality_pts,
+             :num_new_comments
 
   def status
     TaskStatus.status_key_for_name(object.status_name)
   end
+
+  def num_new_comments
+    # TODO: JAKE - Stub using user first -- need to get actual current user
+    object.new_comments_for_user(User.first)
```
Comment (macite) [inline app/serializers/task_serializer.rb]: I suggest delete this method from Task, and have this return a random number...

File: app/models/task.rb (hunk 1 above)
Comment (alexcu) [inline app/models/task.rb]: Fixed in 8a3e1e6

File: app/models/task.rb (hunk 3 above)
Comment (alexcu) [inline app/models/task.rb]: Ok

**Comment by alexcu** [comment]
@jakerenzella Did you end up working from this fork for the new endpoints?

**Comment by alexcu** [comment]
If so I can close this and you can send through a PR for that instead.

**Pull request merged**